.github/workflows/publish.yml

```diff
@@ -2,10 +2,6 @@
 name: Publish
 on:  # yamllint disable-line rule:truthy
   push:
-    branches:
-      - "master"
```

Comment:
Is this OK with ztests? Don't they depend on images being pushed to dockerhub for the master branch?

Reply:
@milan-zededa Good point. But I don't know the "laguage" semantics for github actions; whether it is an "and" operation or an "or" operation between the branches and the tags. Let's assume it is an "or" and I'll add master to it and then we will see when we build the next 9.4.x,

Reply:
Yes, you're correct they use or operation to match patterns to run workflow on branches [link](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-including-branches)

Reply:
There is either a typo in the 4th paragraph in https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushbranchestagsbranches-ignoretags-ignore (regarding "paths") or it says if both are defiined it needs to satisfy both.

Also, I don't know what the tag is really matching - that the tag exists? (all of the release tags exist in the master branch and also in the release branches). What we really want is to only run on the release branches when the tag was just added. 

Maybe this PR is all pointless?
What is breaking is when we run on e.g., 10.4 branch but there is no 10.4.0 tag yet (because the tag was added a millisecond after the branch was created).

Reply:
For that we can use
```
on:
  create:
    tags:
      - '*'
 ```
 We can actually set-up wildcard to match tags we want (i.e. *.\d\.*.\d) 
 And then once tag is created this workflow will run. Only problem is security, but I think there's no way for us to expose secrets that way

Reply:
Nothing should change with respect to the secrets just because we change the "on" part. 
If we can have "on create tags" or'ed with "on push branch master" then we can make sure we publish when a new X.Y.Z tag is added. Is such an "or" on on statements possible? Or do we need to have two separate workflows - one which runs on tag creation and a separate which runs on push to master?